packages/aeson/src/Conferer/Source/Aeson.hs

```diff
@@ -154,3 +178,31 @@ resultToMaybe :: Result a -> Maybe a
 resultToMaybe (Error _) = Nothing
 resultToMaybe (Success a) = Just a
 
+type RawKey = [Text]
+
+-- | Validates that a json has the correct format for keys,
+-- since Conferer 'Key's are pretty restricted.
+--
+-- The Source will work with incorrect keys, it will ignore
+-- them which is not intuitive for the user.
+validateJsonKeys :: Value -> [RawKey]
+validateJsonKeys = go []
```

Comment:
I have 2 suggestions about this function:
- Is it possible to split the logic into finding all raw keys with subkeys on one side and then filtering the invalid subkeys? I think that might make it a bit easier to follow.
- If the list is empty it means no errors and if the list has elements it means there are invalid keys, right? Can we change the return type to an `Either`? if not, can we change the name of the function to something like `invalidJsonKeys` ?